design/open/gc-on-parquet.md

```diff
@@ -0,0 +1,25 @@
+## Garbage collection using Parquet format
+
+### Motivation
+
+Garbage collection in lakeFS is struggling at large scale.
+
+### Why Parquet?
+
+Parquet is more than a file format. It's a breathing open-source project aiming to provide exactly what we're looking for here: efficient data retrieval. GC is essentially a huge anti-join, and it may be very difficult to achieve a similar optimization in our dedicated format.
+
+
+### Steps
+
+#### 1. Sync
+Ranges of the repository are translated into Parquet files. These are stored in a new metadata location under the repository's storage namespace.
+
+#### 2. Sanity
+Verify all the ranges required for this run are present, and that the number of entries in each range is correct.
+
+#### 3. Start
+Do an anti-join extracting all the addresses that need to be deleted.
+
+### Considerations
+1. Storage: this requires using more of the user's storage. I don't think that's an issue since metadata is usually very small compared to the data size. In the future we can also optimize the set of ranges we actually bring there.
```

Comment:
This will essentially double storage used for metadata.  (Pretty much all uncompressible data in a range is the random part of the object address, so a column store cannot do much more than the RocksDB / LevelDB file format.)